src/helpers/ci.ts

```diff
@@ -48,6 +48,10 @@ export const PROVIDER_TO_DISPLAY_NAME = {
   buddy: 'Buddy',
 }
 
+// DD_GITHUB_JOB_NAME is an override that is required for adding custom tags and metrics
+// to GHA jobs if the 'name' property is used. It's ok for it to be missing in case the name property is not used.
```

Comment:
❓  It's this kind of 'name' that you are talking about, right?
https://github.com/DataDog/datadog-ci/blob/c0a0cc657a47682106d9fbf691b8c282ea0404e0/.github/workflows/ci.yml#L7-L14

Are you saying that, in the end, the job name is totally optional? i.e. that having no `name` and no `DD_GITHUB_JOB_NAME` is acceptable?

Reply:
The name is in `GITHUB_JOB` this is an optional workaround for when they are providing a custom name like in your example

Reply:
Correct. The `GITHUB_JOB` environment variable will contain `build-and-test` both when a name is specified and when it isn't. In order to support the case in which is specified we will need customers to specify another env var to point to the custom name

Reply:
Oh ok, I see. Just want to point out that there is the same issue when using a job `matrix`: although GitHub's UI shows the right name (with [`${{ matrix.version }}`](https://github.com/DataDog/datadog-ci/blob/c0a0cc657a47682106d9fbf691b8c282ea0404e0/.github/workflows/ci.yml#L12) between parenthesis, see screenshot), the `GITHUB_JOB` stays the same.

Would be worth to document that `DD_GITHUB_JOB_NAME` is a good use for that as well:
```yaml
env:
  DD_GITHUB_JOB_NAME: "Build and test (${{ matrix.version }})"
```

<img width="493" alt="image" src="https://github.com/DataDog/datadog-ci/assets/9317502/874ac488-c294-46c6-bf18-35ba3506d2c2">

Reply:
Oh, this is a great spot thanks. I will make sure to include this in the docs I'll create